src/org/labkey/test/tests/SampleSetTest.java

```diff
@@ -617,6 +646,81 @@ public void deleteLineageParent() throws IOException, CommandException
         dgen.deleteDomain(createDefaultConnection(true));
     }
 
+
+    @Test
+    public void testStringLookupFields() throws IOException, CommandException
+    {
+        String sampleSetName = "10000Samples";
+        String listName = "MainList";
+
+        goToProjectHome();
+        new PortalHelper(this).addWebPart("Lists");
+
+        log("Creating the sample set of 10000 samples");
+        FieldDefinition.LookupInfo lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "exp.materials", sampleSetName);
+        TestDataGenerator dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        TestDataGenerator.simpleFieldDef("name", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("label", FieldDefinition.ColumnType.String)));
+        dgen.addDataSupplier("label", () -> dgen.randomString(10))
+                .withGeneratedRows(10000);
+        dgen.createDomain(createDefaultConnection(true), "SampleSet");
+        dgen.insertRows(createDefaultConnection(true), dgen.getRows());
+
+        log("Waiting for the sample data to get generated");
+        goToProjectHome();
+        waitAndClickAndWait(Locator.linkWithText(sampleSetName));
+        waitForTextWithRefresh(500000, "1 - 100 of 10,000");
```

Comment:
Is it necessary to go to this page only to go somewhere else immediately after all this waiting?  It looks like the insertRows command is a blocking command so it won't return until the samples have been inserted.  Is that not the case? Or does this cause the samples to get cached so other operations can proceed without waiting?

Reply:
There was no other option that i explored to verify if the rows were generated unless i see the number in sample set. 

You made a good point, i rearranged some code so that sample set is visited just once both to verify 10000 records and adding a additional row.

Reply:
A more expedient way to verify that the rows have been inserted is to capture the response from insertRows and make sure there isn't an error there and/or that the number of rows affected is as expected.  Then you can bypass this very costly waiting for the UI altogether.